File: app/src/main/java/com/botijasoftware/utils/VertexBufferDeclaration.java
package com.botijasoftware.utils;

import java.util.ArrayList;

public class VertexBufferDeclaration {
	protected ArrayList<VertexBufferDefinition> declaration = new ArrayList<VertexBufferDefinition>();

	public int add(VertexBufferDefinition vbd) {
		declaration.add(vbd);
		return vbd.mSize * vbd.getDataSize();
	}

	public int getCount() {
		return declaration.size();
	}

	public VertexBufferDefinition getDefinition(int index) {
		if (index >= 0 && index < declaration.size()) {
			return declaration.get(index);
		}
		return null;
	}
	
	public VertexBufferDefinition searchByUse(int use) {
		for (int i = 0; i < declaration.size(); i++) {
			VertexBufferDefinition vbd = declaration.get(i);
			if (vbd.mUsage == use)
				return vbd;
		}
		return null;
	}

}

File: app/src/main/java/com/botijasoftware/utils/VertexBufferDeclaration.kt
package com.botijasoftware.utils

import java.util.ArrayList

class VertexBufferDeclaration {
    protected var declaration = ArrayList<VertexBufferDefinition>()

    fun add(vbd: VertexBufferDefinition): Int {
        declaration.add(vbd)
        return vbd.mSize * vbd.dataSize
    }

    val count: Int
        get() = declaration.size

    fun getDefinition(index: Int): VertexBufferDefinition? {
        if (index >= 0 && index < declaration.size) {
            return declaration[index]
        }
        return null
    }

    fun searchByUse(use: Int): VertexBufferDefinition? {
        for (i in declaration.indices) {
            val vbd = declaration[i]
            if (vbd.usageGL == use)
                return vbd
        }
        return null
    }

}